upload-distribution.sh

```diff
@@ -16,21 +16,40 @@ fi
 
 echo "#####################################################"
 echo "# Uploading Hibernate $PROJECT $RELEASE_VERSION on"
-echo "# SourceForge: $DIST_PARENT_DIR"
+echo "# SourceForge: $REMOTE_DIST_PARENT_DIR"
 echo "#####################################################"
 echo "Workspace: $WORKSPACE"
 
 pushd $WORKSPACE
 
-((echo mkdir $DIST_PARENT_DIR/$RELEASE_VERSION; echo quit) | sftp -b - frs.sourceforge.net) || echo "Directory already exists. Skipping creation."
-scp README.md frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
-scp changelog.txt frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
-scp distribution/target/hibernate-$PROJECT-$RELEASE_VERSION-dist.zip frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
-scp distribution/target/hibernate-$PROJECT-$RELEASE_VERSION-dist.tar.gz frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
+((echo mkdir $REMOTE_DIST_PARENT_DIR/$RELEASE_VERSION; echo quit) | sftp -b - frs.sourceforge.net) || echo "Directory already exists. Skipping creation."
 
-MODULE=modules/target/hibernate-$PROJECT-modules-$RELEASE_VERSION-wildfly-10-dist.zip
-if [ -f $MODULE ]; then
-	scp $MODULE frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
+REMOTE_DIST_URL=frs.sourceforge.net:$REMOTE_DIST_PARENT_DIR/$RELEASE_VERSION/
+
+scp README.md $REMOTE_DIST_URL
+scp changelog.txt $REMOTE_DIST_URL
+
+# Recursive upload of the dist directory (whose content is project-specific)
+DIST_DIR=distribution/target/dist
+if [ -d $DIST_DIR ]; then
+	# Cd to the dist directory to prevent scp from uploading a "dist" directory
+	pushd $DIST_DIR
+	scp -r . $REMOTE_DIST_URL
+	popd
+fi
+
+# Legacy behavior with explicit uploads - useful for older branches
+LEGACY_DIST_ZIP=distribution/target/hibernate-$PROJECT-$RELEASE_VERSION-dist.zip
+if [ -f $LEGACY_DIST_ZIP ]; then
+	scp $LEGACY_DIST_ZIP $REMOTE_DIST_URL
+fi
+LEGACY_DIST_TAR=distribution/target/hibernate-$PROJECT-$RELEASE_VERSION-dist.tar.gz
+if [ -f $LEGACY_DIST_TAR ]; then
+	scp $LEGACY_DIST_TAR $REMOTE_DIST_URL
+fi
+LEGACY_MODULE=modules/target/hibernate-$PROJECT-modules-$RELEASE_VERSION-wildfly-10-dist.zip
+if [ -f $LEGACY_MODULE ]; then
+	scp $LEGACY_MODULE $REMOTE_DIST_URL
```

Comment:
I would make it a for loop on `modules/target/hibernate-$PROJECT-modules-$RELEASE_VERSION-wildfly-*-dist.zip`.

Reply:
We can add that in another PR if you want, but that's definitely not the purpose of this PR... ? I guess you're concerned about older branches?

Note that for newer branches you can simply put everything in the dist/ directory and it'll work as a charm. Also, it will allow to handle patch files in Validator.
I'll submit PRs to the master branch of each project to illustrate. Here is one for Search: https://github.com/hibernate/hibernate-search/pull/1603